models/external_user.go

```diff
@@ -0,0 +1,34 @@
+package models
+
+import (
+	"time"
+
+	"gorm.io/gorm"
+)
+
+// ExternalUser contains external user information from OAuth providers.
+type ExternalUser struct {
+	CreatedAt   time.Time
+	UpdatedAt   time.Time
+	DeletedAt   gorm.DeletedAt `gorm:"index"`
+	ExternalID  string         `gorm:"primaryKey"`
+	UserID      uint           `gorm:"primaryKey"`
+	User        User
+	Provider    string
+	Email       string
+	Username    string
+	ExternalURL string
+	AccessToken string
+	RawData     string
+}
```

Comment:
Do we need to store anything else? I added raw data should we ever need to add something later on, but the plan is to eventually (once we have more OAuth providers) get rid of it.